GliaWidgetsTests/Sources/CallVisualizer/Coordinator/CallVisualizerCoordinatorTests.swift

```diff
@@ -0,0 +1,141 @@
+
+    func test_showVisitorCodeViewController() throws {
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        coordinator.showVisitorCodeViewController(by: .alert(viewController))
+        
+        XCTAssertTrue(viewController.presentedViewController is CallVisualizer.VisitorCodeViewController)
+    }
+
+    func test_handleAcceptedUpgrade() {
+        var calledEvents: [CallVisualizer.Coordinator.DelegateEvent] = []
+        coordinator.environment.eventHandler = { calledEvents.append($0) }
+        coordinator.handleAcceptedUpgrade()
+
+        XCTAssertTrue(calledEvents.contains(.maximized))
+    }
+
+    func test_handleEngagementRequestAccepted() throws {
+        let site = CoreSdkClient.Site(
+            id: .mock, defaultOperatorPicture: nil,
+            alwaysUseDefaultOperatorPicture: false,
+            allowedFileSenders: try .mock(),
+            maskingRegularExpressions: [],
+            visitorAppDefaultLocale: "",
+            mobileConfirmDialogEnabled: false,
+            mobileObservationIndicationEnabled: true,
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        let answer = Command<Bool> { boolean in
+            XCTAssertTrue(boolean)
+        }
+
+        coordinator.handleEngagementRequestAccepted(answer)
+    }
+
+    func test_handleEngagementRequestAcceptedMobileConfirmDialogEnabled() throws {
+        let scene = try XCTUnwrap(UIApplication.shared.connectedScenes.first as? UIWindowScene)
+        let window = scene.windows.first
+        let oldRootViewController = window?.rootViewController
+        window?.rootViewController = viewController
+        defer { window?.rootViewController = oldRootViewController }
+
+        let presenter = CallVisualizer.Presenter(presenter: { self.viewController })
+        coordinator.environment.presenter = presenter
+
+        let site = CoreSdkClient.Site(
+            id: .mock, defaultOperatorPicture: nil,
+            alwaysUseDefaultOperatorPicture: false,
+            allowedFileSenders: try .mock(),
+            maskingRegularExpressions: [],
+            visitorAppDefaultLocale: "",
+            mobileConfirmDialogEnabled: true,
+            mobileObservationIndicationEnabled: true,
+            mobileObservationVideoFps: try videoFps(),
+            mobileObservationEnabled: true
+        )
+
+        coordinator.environment.fetchSiteConfigurations = { callback in
+            callback(.success(site))
+        }
+
+        let answer = Command<Bool> { boolean in
+            XCTAssertTrue(boolean)
```

Comment:
Same idea about asserting in callback.

Reply:
I changed it here, because I realized that indeed this assertion wasn't even running, but the problem is that to run it, I need to manually press the AlertViewController because the answer command depends on what the user presses. Thus I believe that because showing the AlertViewController is what the coordinator takes care of, then that should be enough for the test. 21298a0a635189b8bb2191b3dc1b34e4e621e7be let me know if it makes sense.

Reply:
Make sense.